src/compiler/types.ts

```diff
@@ -5244,6 +5244,7 @@ namespace ts {
         pattern?: DestructuringPattern;  // Destructuring pattern represented by type (if any)
         aliasSymbol?: Symbol;            // Alias associated with type
         aliasTypeArguments?: readonly Type[]; // Alias type arguments (if any)
+        aliasMapper?: TypeMapper | undefined; // Mapper used to instantiate alias (if any)

```

Comment:
We need the mapper used to instantiate the alias so we can rehydrate the values of the alias's outer type parameters at comparison-time later (realistically, we only compare variance for object and conditional types, which both have a `.mapper` member already - this should be a clone of it in the simple cases, only diverging when odd union/intersection re-aliasing occurs).